test/integration/handlers/authorize-handler_test.js

```diff
@@ -653,25 +636,23 @@ describe('AuthorizeHandler integration', function() {
       handler.getClient(request).should.be.an.instanceOf(Promise);
     });
 
-    describe('with `client_id` in the request body', function() {
-      it('should return a client', function() {
-        var client = { grants: ['authorization_code'], redirectUris: ['http://example.com/cb'] };
-        var model = {
-          getAccessToken: function() {},
-          getClient: function() {
-            return client;
-          },
-          saveAuthorizationCode: function() {}
-        };
-        var handler = new AuthorizeHandler({ authorizationCodeLifetime: 120, model: model });
-        var request = new Request({ body: { client_id: 12345, response_type: 'code' }, headers: {}, method: {}, query: {} });
-
-        return handler.getClient(request)
-          .then(function(data) {
-            data.should.equal(client);
-          })
-          .catch(should.fail);
+    it('should support callbacks', function() {
+      var model = {
+        getAccessToken: function() {},
+        getClient: function(clientId, callback) {
```

Comment:
Changing `getClient` to check that the callback is always passed as third argument would catch the problem described in [this comment](https://github.com/oauthjs/node-oauth2-server/pull/272/files?diff=split#r65514207).

```js
getClient: function(clientId, clientSecret, callback) {
  should(clientSecret).equal(null);
  callback(null, { grants: ['authorization_code'], redirectUris: ['http://example.com/cb'] });
}
```